src/app/admin/admin-issuer-edit/admin-issuer-edit.component.ts

```diff
@@ -63,38 +62,48 @@ export class AdminIssuerEditComponent {
 
   updateNameAndLogo(issuer: IssuerWithInfo) {
     return () => {
-      if (this.updateForm.value.name !== issuer.name || this.updateForm.value.logo?.[0]) {
-        return this.issuerService.uploadInfo(
-          this.updateForm.value.name,
-          this.updateForm.value.logo?.[0],
-          issuer,
-        ).pipe(
-          switchMap(uploadRes => this.issuerService.updateInfo(issuer.contractAddress, uploadRes.path)),
-          tap(() => this.routerService.navigate(['/../admin'])),
-          switchMap(() => this.dialogService.info('Issuer successfully updated!', false)),
-        )
-      }
-
-      return fromPromise(this.routerService.navigate(['/../admin']))
+      return this.issuerService.uploadInfo(
+        this.updateForm.value.name,
+        this.updateForm.value.logo?.[0],
+        issuer,
+      ).pipe(
+        switchMap(uploadRes => this.issuerService.updateInfo(issuer.contractAddress, uploadRes.path)),
+        switchMap(() => this.dialogService.info('Issuer successfully updated!', false)),
+        tap(() => this.refreshIssuer()),
+        tap(() => this.routerService.navigate(['/admin'])),
+      )
     }
   }
 
   updateWalletApproverAddress(issuer: IssuerWithInfo) {
-    return () => this.issuerService.changeWalletApprover(
-      issuer.contractAddress, this.updateWalletApproverAddressForm.value.walletApproverAddress,
-    ).pipe(
-      tap(() => this.updateWalletApproverAddressForm.markAsPristine()),
-      switchMap(() => this.dialogService.info('Wallet approver changed successfully!', false)),
-    )
+    return () => {
+      return this.issuerService.changeWalletApprover(
+        issuer.contractAddress, this.updateWalletApproverAddressForm.value.walletApproverAddress,
+      ).pipe(
+        tap(() => this.updateWalletApproverAddressForm.markAsPristine()),
+        switchMap(() => this.dialogService.info('Wallet approver changed successfully!', false)),
+      )
+    }
   }
 
   updateOwnerAddress(issuer: IssuerWithInfo) {
-    return () => this.issuerService.changeOwner(
-      issuer.contractAddress, this.updateOwnerAddressForm.value.ownerAddress,
-    ).pipe(
-      tap(() => this.updateOwnerAddressForm.markAsPristine()),
-      switchMap(() => this.dialogService.info('Owner changed successfully!', false)),
-    )
+    return () => {
+      return this.issuerService.changeOwner(
+        issuer.contractAddress, this.updateOwnerAddressForm.value.ownerAddress,
+      ).pipe(
+        switchMap(() => this.dialogService.info('Owner changed successfully!', false)),
+        tap(() => this.refreshIssuer()),
+        tap(() => this.routerService.navigate(['/'])),
```

Comment:
the issuer must be refreshed (so the admin tab goes away) and the user must be navigated somewhere else as owner has definitely changed to somebody else.